docs/make.jl

```diff
@@ -9,7 +9,7 @@ makedocs(;
     repo="https://github.com/JuliaObjects/ConstructionBase.jl/blob/{commit}{path}#L{line}",
     sitename="ConstructionBase.jl",
     authors="Takafumi Arakaki, Rafael Schouten, Jan Weidner",
-    assets=String[],
+    strict=true,
```

Comment:
So this means only build the docs if doctests pass?

Reply:
Yes (or I think more precisely deploy only if doctests pass)

Reply:
Ok thanks I was not aware that this was off by default.

Reply:
In Documenter 0.24 it looks like we'll have `Documenter.doctest` so we can run doctests more easily via runtests.jl (and stop worrying about a setup like this).